envi/archs/arm/const.py

```diff
@@ -3,6 +3,32 @@
 MODE_JAZELLE    = 2
 MODE_THUMBEE    = 3
 
+'''
+bitfields to support different versions
+For simplicity will use 0xFFFF to say all.
+Note that thumb and arm are both represented
+thumbee is shown but may never be fully implimented
+since was already depreciated
+'''
+archBitMask = [
+#    architecture   bitmask              dec     hex
+    ('THUMB16',     0b0000000000000001), # 1      1
```

Comment:
i believe thumb16, thumb2, and thumbee are all defined by the ARMv* items below.  for instance, ARMv5T includes THUMB16.  we should be able to clearly define an architecture using the ARMv* only.  for example, ARMv7M will be THUMB2 only.  ARMv7AR will support THUMB2.  if you'd like THUMB2 to be easily available, make it a bitfield combination of all ARMv* architectures which support THUMB2.

REV_ARMv7 = (REV_ARMv7M | REV_ARMv7AR)
REV_THUMB2 = (REV_ARMv6T2 | REV_ARMv7 | REV_ARMv8)